src/GitTabExpansion.ps1

```diff
@@ -380,16 +389,21 @@ function GitTabExpansionInternal($lastBlock, $GitStatus = $null) {
             gitAddFiles $GitStatus $matches['files']
         }
 
+        # Handles git checkout -- <path>
+        "^checkout.* -- (?<files>\S*)$" {
+            gitCheckoutFiles $GitStatus $matches['files']
+        }
+
         # Handles git restore -s <ref> - must come before the next regex case
-        "^restore.* -s\s*(?<ref>\S*)$" {
+        "^restore.* (?-i)-s\s*(?<ref>\S*)$" {
             gitBranches $matches['ref'] $true
             gitTags $matches['ref']
             break
         }
 
-        # Handles git checkout -- <path> and git restore <path>
-        "^(?:checkout.* --|restore.*) (?<files>\S*)$" {
-            gitCheckoutFiles $GitStatus $matches['files']
+        # Handles git restore <path>
+        "^restore(?:.* (?<staged>(?:-S|--staged))|.*) (?<files>\S*)$" {
```

Comment:
This needs the case sensitive check but only for the `-S` which would require an additional non-capturing group. Not a big deal other than readability. Other than that LGTM

[ I posted this comment over a week ago, but apparently I hit 'start a review' instead of add comment. I noticed this morning that GitHub was telling me that my comment was still pending(?). I cancelled it and re-posted as a comment. I admit that I don't use GitHub much (we use Azure git) so I'm not too familiar with the UI. Was that the *right* thing to do? ]